sympy/printing/printer.py

```diff
@@ -381,5 +379,11 @@ def __signature__(self) -> inspect.Signature:
 def print_function(print_cls):
     """ A decorator to replace kwargs with the printer settings in __signature__ """
     def decorator(f):
-        return _PrintFunction(f, print_cls)
+        if sys.version_info < (3, 9):
+            # We have to create a subclass so that `help` actually shows the docstring in older python versions.
+            # IPython and Sphinx do not need this, only a raw python console.
+            cls = type(f'{f.__qualname__}_PrintFunction', (_PrintFunction,), dict(__doc__=f.__doc__))
```

Comment:
```suggestion
            cls = type(f'_PrintFunction', (_PrintFunction,), dict(__doc__=f.__doc__))
```
(hiding the class name since the subclass is a hack)

Reply:
I guess I'm not keen on hiding the fact that we have this hack. It makes the debugging harder. 

By the way, I hope this dynamically created subclass doesn't cause issues with things like pickling. If it does, then I think the whole thing really isn't worth it, and we should just use the non-updating version of assigning `__signature__` to the function.

Reply:
So, there are two different "hacks" here:

* The existence of `_PrintFunction`, which is to ensure that `latex` actually respects the global settings and that `inspect.signature` shows them. Previously we did only the former, then someone didn't like the documentation was broken and fixed the latter at the expense of breaking the former.
* The existence of the subclass in Python < 3.8, which is just to fix the builtin `help`

IMO, the former hack is interesting, and the latter is not - so it's reasonable to try and hide the parts of the latter hack which are just noise.

> I hope this dynamically created subclass doesn't cause issues with things like pickling

Looks like I already broke that. It's easy to fix with:
```
def __reduce__(self):
    return self.__wrapped__.__name__
```

Reply:
xref #20468, where I fix pickling